### ISSUE TITLE
Get topic for train docs

### ISSUE THREAD

**Issue by sadhana01** [bab2min/tomotopy#90]
I would like to classify the train documents into their respective topics and I tried model.infer and it gave me a type mismatch error for 'doc'. Is there a better way of doing this ?

```
for doc_words in train_words_lemmatized:
     doc_inst = model.make_doc(doc_words)
     topic_dist, ll = model.infer(doc_inst)
```
The code above gives the following error:
```    
topic_dist, ll = model.infer(doc_inst)
Exception: `doc` must be tomotopy.Document type or list of tomotopy.Document
```

But I ascertained that the type of doc_inst is <class 'tomotopy.Document'>

train_words_lemmatized is a list of lists of words for each document tokenized and lemmatized, in the same format I added them to the train model with model.add_doc()
model is defined as follows:
`model = tp.HDPModel(tw=tp.TermWeight.IDF, min_cf=5, rm_top=5)`

Any help would be appreciated , thanks

**Comment by bab2min** [comment]
```Exception: `doc` must be tomotopy.Document type or list of tomotopy.Document``` This exception seems to be a bug that appeared in version 0.10.0. You can avoid this bug using ```topic_dist, ll = model.infer([doc_inst])``` instead of ```topic_dist, ll = model.infer(doc_inst)```.

And actually for train documents, you don't need to re-infer them. You can just access their topic distribution like:
```
model.docs[0].get_topic_dist() # returns topic distribution of train document #0
model.docs[42].get_topic_dist() # returns topic distribution of train document #42
```

**Comment by andrewmsilva** [comment]
I'm with a similar problem: I save the model with `full=False` and need to get the topic distribution from the train documents when I reload the model. However, as `full=False` was used, I guess `.infer(doc)` is necessary. Do you know a way to avoid re-inferring the data and get the topic distribution?

**Comment by bab2min** [comment]
Hi @andrewmsilva,
Topic assignments of words in training documents is only saved when `full=True`. Unfortunately, if you saved the model with `full=False`, there is no way to restore topic assignments of training docs, so you cannot avoid re-inferring. The only way to avoid re-inferring is to save the model with `full=True`.

**Comment by bab2min** [comment]
The bug (https://github.com/bab2min/tomotopy/issues/90#issuecomment-760185952) has been fixed at v0.10.1.
